pkg/controller/jobframework/reconciler.go

```diff
@@ -558,10 +598,8 @@ func (r *JobReconciler) startJob(ctx context.Context, job GenericJob, object cli
 // stopJob will suspend the job, and also restore node affinity, reset job status if needed.
 // Returns whether any operation was done to stop the job or an error.
 func (r *JobReconciler) stopJob(ctx context.Context, job GenericJob, object client.Object, wl *kueue.Workload, eventMsg string) error {
-	info := getPodSetsInfoFromWorkload(wl)
```

Comment:
this change worries me a little. This operation is expensive, as it does a deep copy of multiple maps.

Prefer to pass a boolean to Stop to slightly change the behavior, if necessary.